server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconcilerMetricsIT.java

```diff
@@ -68,6 +69,7 @@ public void testDesiredBalanceMetrics() {
         final var infoService = (InternalClusterInfoService) internalCluster().getCurrentMasterNodeInstance(ClusterInfoService.class);
         ClusterInfoServiceUtils.setUpdateFrequency(infoService, TimeValue.timeValueMillis(200));
         assertNotNull("info should not be null", ClusterInfoServiceUtils.refresh(infoService));
```

Comment:
It seems to me that the asyncRefresh for clusterInfo can also pause which makes the `refresh` call here to get back a stale result since the newRefresh is [short-circuited](https://github.com/elastic/elasticsearch/blob/27eb01356be34742a12d207545fca7af01573e93/server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java#L364-L365) if there is already one running. If that does happen, it might also lead to test failure. But in practice, I was not able to make it happen, probably due to the short update interval of 200ms. So I am leaving it as is.